### DISCUSSION TITLE
Binary Ninja running out of memory during analysis

### DISCUSSION THREAD

**Discussion by nonnumericdave** [Vector35/binaryninja-api#4683] [General] [answered]
I'm using Binary Ninja 3.5 on MacOS on an M1 Max with 64GB of memory, and I've noticed that during the analysis of a binary, it is causing my system to run out of memory.  In Activity Monitor, I see the virtual memory memory usage get up to over 200 GB.

I don't want to say the binary in question (you can contact me privately for more details - it's a binary you definitely have access to).

Any ideas?

**Comment by alexrp** [comment]
Assuming this isn't a bug (I'm not qualified to say), my first thought would be to check this setting:

![image](https://github.com/Vector35/binaryninja-api/assets/44076/31532eaa-8732-4da0-914b-599947e003e3)

Depending on the size of the binary, setting this to a non-default value can result in an explosion in memory usage.

**Comment by psifertex** [comment]
Shoot us an email binaryninja at vector35 dotcom with a bit more details but I will say we test with a debug chrome and it manages to fit in 64gb and is fairly massive. That said, we can go heavy on memory depending on some other factors. 

Disabling linear sweep and adjusting several limits and timeouts in the open with options analysis setting can help in general.

**Comment by 0cyn** [comment]
Resolved in development **builds >= 3.6.4596**

**Comment by fuzyll** [comment, marked as answer]
This was actually a bug on our end, the same root cause as #3885. As per @0cyn's comment, this should be resolved in builds >= 3.6.4596.
